src/app/public/modules/lookup/lookup-show-more-modal.component.ts

```diff
@@ -0,0 +1,331 @@
+  styleUrls: ['./lookup-show-more-modal.component.scss'],
+  providers: [SkyDataManagerService],
+  changeDetection: ChangeDetectionStrategy.OnPush
+})
+export class SkyLookupShowMoreModalComponent implements OnDestroy, OnInit {
+
+  /**
+   * @internal
+   * Fires when users select the "Add" button
+   */
+  @Output()
+  public addClick: EventEmitter<void> = new EventEmitter();
+
+  public items: any[];
+
+  public dataManagerConfig = {
+    sortOptions: [
+      {
+        id: 'az',
+        label: 'Alphabetical (A - Z)',
+        descending: false,
+        propertyName: 'name'
+      },
+      {
+        id: 'za',
+        label: 'Alphabetical (Z - A)',
+        descending: true,
+        propertyName: 'name'
+      }
+    ]
+  };
+
+  public dataState = new SkyDataManagerState({});
+  public displayedItems: any[] = [];
+  public itemsHaveMore: boolean = false;
+  public viewId = 'repeaterView';
+  public viewConfig: SkyDataViewConfig = {
+    id: this.viewId,
+    name: 'Repeater View',
+    icon: 'list',
+    searchEnabled: true,
+    filterButtonEnabled: false,
+    multiselectToolbarEnabled: this.context.selectMode === 'multiple',
+    onClearAllClick: this.clearAll.bind(this),
+    onSelectAllClick: this.selectAll.bind(this)
+  };
+
+  private itemIndex: number = 0;
+  private ngUnsubscribe = new Subject();
+
+  constructor(
+    public modalInstance: SkyModalInstance,
+    public context: SkyLookupShowMoreContext,
+    private changeDetector: ChangeDetectorRef,
+    private dataManagerService: SkyDataManagerService
+  ) { }
+
+  public ngOnInit(): void {
+    this.dataState.searchText = this.context.initialSearch;
+    this.dataManagerService.initDataView(this.viewConfig);
+
+    this.dataManagerService.initDataManager(
+      {
+        activeViewId: 'repeaterView',
+        dataManagerConfig: this.dataManagerConfig,
+        defaultDataState: this.dataState
+      }
+    );
+
+    setTimeout(() => {
+      this.addItems();
+
+      this.dataManagerService.getDataStateUpdates(this.viewId).pipe(
+          debounceTime(250),
+          takeUntil(this.ngUnsubscribe)
+        ).subscribe(state => {
+        if (this.dataState.searchText !== state.searchText) {
+          this.itemIndex = 10;
+        }
+        this.dataState = state;
+        this.updateData();
+      });
+    });
+  }
+
+  public ngOnDestroy(): void {
+    this.ngUnsubscribe.next();
+    this.ngUnsubscribe.complete();
+  }
+
+  public addButtonClicked(): void {
+    this.addClick.emit();
+  }
+
+  public addItems(): void {
+    if (!this.items || this.items.length === 0) {
+      let selectedIds: any[] = this.dataState.selectedIds?.slice() || [];
+      const initialIsArray = Array.isArray(this.context.initialValue);
+      this.items = this.context.items?.map(item => {
+        return {
+          value: item,
+          selected: false
+        };
+      });
+      this.items.forEach(item => {
+        if ((this.isEquivalent(this.context.initialValue, item.value)) ||
```

Comment:
The logic in this `if` block is a bit difficult to parse out, personally. What if we broke it out into separate statements to make it easier to read?

Reply:
I split this out a little more in the `lookup-modals` PR.